tiny_cnn/layers/convolutional_layer.h

```diff
@@ -233,12 +238,26 @@ class convolutional_layer : public feedforward_layer<Activation> {
      * @param out_data     output vectors
      **/
     void forward_propagation(const std::vector<tensor_t*>& in_data,
-                             std::vector<tensor_t*>&       out_data) {
+                             std::vector<tensor_t*>&       out_data) { 
+        // forward convolutional op context
+        auto ctx = new OpKernelContext(in_data, out_data);
+             ctx->setParams(&params_);
+             ctx->setParallelize(layer::parallelize());
+
```

Comment:
Memory leak occurs on this line. If ```ctx``` can be destructed at the end of this scope, we can simply use stack.
```
OpKernelContext ctx(in_data, out_data);
```